apis/python/src/tiledbsoma/options/_tiledb_create_options.py

```diff
@@ -1,170 +1,301 @@
 
-    def dim_filters(
-        self, dim: str, default: Sequence[StrOrMap] = ()
-    ) -> Sequence[tiledb.Filter]:
-        return _build_filters(self._dim(dim).get("filters", default))
+    def validity_filters_tiledb(self) -> Optional[Tuple[tiledb.Filter, ...]]:
+        """Constructs the real TileDB Filters to use for the validity map."""
+        if self.validity_filters is None:
+            return None
+        return tuple(_build_filter(f) for f in self.validity_filters)
 
-    def dim_tile(self, dim: str, default: int = DEFAULT_TILE_EXTENT) -> int:
-        return self._dim(dim).get("tile", default)
+    def dim_filters_tiledb(
+        self, dim: str, default: Sequence[_FilterSpec] = ()
+    ) -> Tuple[tiledb.Filter, ...]:
+        """Constructs the real TileDB Filters to use for the named dimension."""
+        return _filters_from(self.dims, dim, default)
 
-    def attr_filters(
-        self, attr: str, default: Sequence[StrOrMap] = ()
-    ) -> Sequence[tiledb.Filter]:
-        return _build_filters(self._attr(attr).get("filters", default))
+    def dim_tile(self, dim_name: str, default: int = DEFAULT_TILE_EXTENT) -> int:
+        """Returns the tile extent for the given dimension."""
+        try:
+            dim = self.dims[dim_name]
+        except KeyError:
+            return default
+        return default if dim.tile is None else dim.tile
 
-    def _dim(self, dim: str) -> Dict[str, Any]:
-        return cast(Dict[str, Any], self.get("dims", {}).get(dim, {}))
+    def attr_filters_tiledb(
+        self, name: str, default: Sequence[_FilterSpec] = ()
+    ) -> Tuple[tiledb.Filter, ...]:
+        """Constructs the real TileDB Filters to use for the named attribute."""
+        return _filters_from(self.attrs, name, default)
 
-    def _attr(self, attr: str) -> Dict[str, Any]:
-        return cast(Dict[str, Any], self.get("attrs", {}).get(attr, {}))
 
-    # Mapping implementation:
+_T = TypeVar("_T")
 
-    def __getitem__(self, it: str) -> Any:
-        return self._config[it]
 
-    def __len__(self) -> int:
-        return len(self._config)
+def _dig_platform_config(
+    input: object, typ: Type[_T], full_path: Tuple[str, ...]
+) -> Union[Dict[str, object], _T]:
+    """Looks for an object of the given type in dictionaries.
 
-    def __iter__(self) -> Iterator[str]:
-        return iter(self._config)
+    This is used to extract a valid object out of `platform_config`. If an
+    object matching `typ` is found, it is returned directly; otherwise we
+    descend in the next dictionary key specified by `full_path`.
+    """
+    current = input
+    path = full_path
+    while path:
+        if isinstance(current, typ):
+            return current
+        if not isinstance(current, dict):
+            return {}
+        key, path = path[0], path[1:]
+        try:
+            current = current[key]
+        except KeyError:
+            return {}
+    if not isinstance(current, (typ, dict)):
```

Comment:
Will this `isinstance(current, typ)` ever be true here? If not, can just check for the `dict` at this point.